tests/instructionCompiler_test.py

```diff
@@ -0,0 +1,750 @@
+import logging
+
+import pytest
+from cu2qu.ufo import font_to_quadratic
+from fontTools.pens.hashPointPen import HashPointPen
+from fontTools.ttLib.tables._g_l_y_f import (
+    OVERLAP_COMPOUND,
+    ROUND_XY_TO_GRID,
+    USE_MY_METRICS,
+)
+from fontTools.ttLib.ttFont import TTFont
+
+from ufo2ft.instructionCompiler import InstructionCompiler
+
+from .outlineCompiler_test import getpath
+
+TRUETYPE_INSTRUCTIONS_KEY = "public.truetype.instructions"
+
+
+def expect_maxp(
+    font,
+    maxStorage=0,
+    maxFunctionDefs=0,
+    maxInstructionDefs=0,
+    maxStackElements=0,
+    maxSizeOfInstructions=0,
+    maxZones=1,
+    maxTwilightPoints=0,
+):
+    maxp = font["maxp"]
+    assert maxp.maxStorage == maxStorage
+    assert maxp.maxFunctionDefs == maxFunctionDefs
+    assert maxp.maxInstructionDefs == maxInstructionDefs
+    assert maxp.maxStackElements == maxStackElements
+    assert maxp.maxSizeOfInstructions == maxSizeOfInstructions
+    assert maxp.maxZones == maxZones
+    assert maxp.maxTwilightPoints == maxTwilightPoints
+
+
+def get_hash_ufo(glyph, ufo):
+    hash_pen = HashPointPen(glyph.width, ufo)
+    glyph.drawPoints(hash_pen)
+    return hash_pen.hash
+
+
+def get_hash_ttf(glyph_name, ttf):
+    aw, _lsb = ttf["hmtx"][glyph_name]
+    gs = ttf.getGlyphSet()
+    hash_pen = HashPointPen(aw, gs)
+    ttf["glyf"][glyph_name].drawPoints(hash_pen, ttf["glyf"])
+    return hash_pen.hash
+
+
+@pytest.fixture
+def quadfont():
+    font = TTFont()
+    font.importXML(getpath("TestFont-TTF-post3.ttx"))
+    return font
+
+
+@pytest.fixture
+def quadufo(FontClass):
+    font = FontClass(getpath("TestFont.ufo"))
+    font_to_quadratic(font)
+    return font
+
+
+@pytest.fixture
+def quaduforeversed(FontClass):
+    font = FontClass(getpath("TestFont.ufo"))
+    font_to_quadratic(font=font, reverse_direction=True)
+    return font
+
+
+@pytest.fixture
+def testufo(FontClass):
+    font = FontClass(getpath("TestFont.ufo"))
+    del font.lib["public.postscriptNames"]
+    return font
+
+
+class InstructionCompilerTest:
+    # _check_glyph_hash
```

Comment:
Leftover?

Reply:
no, I think it means "the group of test methods below check the `_check_glyph_hash` method"